third_party/nvfuser/csrc/codegen.cpp

```diff
@@ -1913,8 +1914,14 @@ class CudaKernelGenerator : private OptOutConstDispatch {
             .append("[")
             .append(work_buffer_offset)
             .append("]");
-        init_vals.arg(genInline(grouped_grop->initVal(expr_index)));
-
+        auto iv = (grouped_grop->initVal(expr_index));
+        // TODO: why data type doesn't match? e.g. complex<float> <--->
+        // complex<double>
```

Comment:
Python scalar only has double, int64_t and complex double, there is no float, int32 and complex float. PyTorch scalar has the same design as python scalar, so the dtype might be mismatch here. I am fine with the change below, but please update this comment.

Reply:
done. thanks for the background knowledge!